### ISSUE TITLE
S3928 reports the parameter name as '' when the referenced argument type is not resolvable

### ISSUE THREAD

**Issue by StingyJack** [SonarSource/sonar-dotnet#4180]
Split this off of https://github.com/SonarSource/sonar-dotnet/issues/653#issuecomment-803505601  per @pavel-mikula-sonarsource request

> This rule also triggers a false positive in this scenario as well. I had not yet installed the nuget package with the IMyLogger in it and the S3928 pops up, apparently not knowing the parameter list. (do these not analyze the source code?)
![image](https://user-images.githubusercontent.com/15381181/112331863-acaa8380-8c8f-11eb-950f-32504144ba0d.png)

**Comment by pavel-mikula-sonarsource** [comment]
Thank you @StingyJack for catching this issue.

**Note for us:** 
This line doesn't resolve the `nameof` value correctly. Might be possible issue to report to the Roslyn as well.
```
parameterNameValue = analysisContext.SemanticModel.GetConstantValue(argumentExpression);
```

**Comment by andrei-epure-sonarsource** [comment]
I reported the issue to the Roslyn team in https://github.com/dotnet/roslyn/issues/54082 

Thanks @StingyJack .

**Comment by andrei-epure-sonarsource** [comment]
This got fixed in Visual Studion / MSBuild 16.10.1
